ui/src/app/store/machine/utils/common.ts

```diff
@@ -0,0 +1,10 @@
+import type { Machine, MachineDetails } from "app/store/machine/types";
+/**
+ * Wether a machine has a Machine or MachineDetails type.
+ * @param machine - The machine to check
+ * @returns Whether the machine is MachineDetails.
+ */
+export const isMachineDetails = (
+  machine?: Machine | MachineDetails | null
```

Comment:
Shouldn't it be either `Machine | null` or `BaseMachine | MachineDetails | null` since `Machine` is already `BaseMachine | MachineDetails`? Maybe we should rename the types to be a bit clearer what the differences are.

Reply:
Yeah good point, I think `Machine | null` is probably appropriate. That's my fault for copying and not thinking.